pyomo/repn/plugins/gams_writer.py

```diff
@@ -143,56 +231,79 @@ def _print_model( self, model, output_file, solver_capability, var_list,
         for con in model.component_data_objects(Constraint, active=True):
             if con.body.is_fixed():
                 continue
+
+            # Ensure GAMS will not encounter domain violations in presolver
+            # operations at current values, which are None (0) by default
+            # Used to handle log and log10 violations, for example
+            try:
+                value(con.body)
+            except:
+                raise RuntimeError("While evaluating: %s\n\tProblemWriter_gams"
+                                   " encountered an error. Ensure initial"
+                                   " variable values\n\tdo not violate"
+                                   " any domains (are you using log/log10?)"
+                                 % con.name)
```

Comment:
The logger of the value function tracks the full trace of exactly which expression caused the error and goes all the way up to the full constraint expression, so that is already handled there.